File: Base/Workspace/Domain/Base/Workspace/Data/SyncResponse.cs
namespace Allors.Workspace.Data
{
    public class SyncResponseObject
    {
        public string i { get; set; }

        public string v { get; set; }
        public string t { get; set; }

        public object[][] roles { get; set; }
        public string[][] methods { get; set; }
    }

    public class SyncResponse {
        public string userSecurityHash { get; set; }

        public SyncResponseObject[] objects { get; set; }
    }
}

File: dotnet/core/workspace/typescript.legacy/workspace/data/responses/syncresponse.ts
namespace Allors.Data {
    export interface SyncResponseObject {
        i: string;
        v: string;
        t: string;

        roles?: any[][];
        methods?: string[][];
    }

    export interface SyncResponse extends Response {
        userSecurityHash: string;
        objects: SyncResponseObject[];
    }
}